test/conformance/suite_test.go

```diff
@@ -135,8 +137,9 @@ func prepareEnvForGatewayConformanceTests(t *testing.T) (c client.Client, gatewa
 		featureGateFlag,
 		"--anonymous-reports=false",
 	}
-
-	require.NoError(t, testutils.DeployControllerManagerForCluster(ctx, globalLogger, env.Cluster(), args...))
+	cancel, err := testutils.DeployControllerManagerForCluster(ctx, globalLogger, env.Cluster(), nil, args...)
+	require.NoError(t, err)
+	t.Cleanup(func() { cancel() })
```

Comment:
What happens if we directly use `t.Cleanup(cancel)` here?

Reply:
That should work as well but I'm hesitant to make this change since everywhere else we're using this "full" syntax. Do you believe this would be more readable?